packages/wrangler/src/entry.ts

```diff
@@ -20,20 +20,21 @@ export type Entry = { file: string; directory: string; format: CfScriptFormat };
 export async function getEntry(
   args: { script?: string; format?: CfScriptFormat | undefined },
   config: Config,
+  env: Environment,
```

Comment:
`args` already has the value for `env`, and `config` already has the env specifc config, so should we instead resolve the `envRootObj` inside `getEntry` itself?

Reply:
The long term solution is actually to change `normalizeAndValidateConfig()` to accept the current environment name, and to return a suitably setup configuration so that downstream commands do not have to worry about this stuff.
There are a number of places in the code where there is a potential bug where we look in `config` rather than `env` for a field.

Reply:
Ok... I might be missing the connection, but I can dig this for now. Maybe it's better that it's explicitly passed like this.

Reply:
Putting together a PR that will demonstrate this.